odl/space/weighting.py

```diff
@@ -500,7 +500,7 @@ def array(self):
 
     def is_valid(self):
         """Return True if the array is a valid weight, i.e. positive."""
-        return np.all(np.greater(self.array, 0))
+        return (self.array > 0).all()
```

Comment:
Again assuming a certain API of the arrays. Perhaps better to have wrapper `all`, `any` and such. Dunno.